node/MonitoringAgentProposalStatus_20211005.MD

```diff
@@ -0,0 +1,72 @@
+# Monitoring Agent - Proposal Status
+
+## Motivation
+
+Somewhere at Mysterium Network lives a Monitoring Agent (MA). Its duty is to do scheduled checks on all node runner proposals in order to verify proposal connectability.
+Currently, this is represented by a bool flag called **monitoring_failed** with 2 possible values: true or false. It turns out - this is not ideal.  
+
+Never before registered proposals will be marked as **monitoring_failed: false** which means that it is not reachable. MA can’t instantly check all brand new proposals (supposedly because it would be expensive to scale it) so instead MA checks each proposal individually and this takes time. At the time of writing - it takes approximately 4 hours for MA to check (~3160) nodes.  
+
+Depending on your luck (as a node runner) you might want to jump into node running business and right off the bat spend4 hours figuring out why your node status reads as failed (marked monitoring_failed: false). Thus bugging Mysterium Network support staff.  
+
+Adding insult to injury, many API consumers down the line (desktop app, mobile app etc..) may filter out such proposals for better user experience which may also make it harder to debug or test the actual availability of your proposals and perhaps hinder your ability to provide an exit node service.  
+
+Finally, MA seemingly doesn't mark the proposal as **monitoring_failed: false** on the first failed attempt. It only does that after it was not able to connect to the specific node within 24h window.  
+
+Bottom line is - bool representation of a complex state does not convey the meaning and adds confusion.
+
+## Solution
+
+In my opinion a much better approach is to have human-readable states. MA knows what is actually going on and so should the proposal API consumers.  
+First we would introduce a new field in the contract. Let’s call that field: **monitoring_state**
+
+### States (only)
+
+This solution would only deal with proposal monitoring states.
+
+#### How It Works
+
+Introduce a new field into Quality Oracle (QO) contract for _/sessions_ endpoint called: **monitoring_state**. This field will contain a more human-readable representation of the current **monitoring_failed** field.
+
+|**monitoring_state**| Description | Can transition to |
+| :--- | :---: | ---:|
+| INITIAL_CHECK_PENDING| Proposal is brand new, waiting for first check by MA | UNHEALTHY, HEALTHY|
+| UNHEALTHY| Proposal has connection issues (for last 24h proposal has been less than 50% reachable by MA) | HEALTHY, UNSTABLE|
+| HEALTHY| Proposal is healthy (for last 24h it has been >75% reachable by MA) | UNHEALTHY, UNSTABLE|
+| UNSTABLE| Proposal is unstable (for last 24h it has been reachable 50%-75% of the time in MA checks) | HEALTHY, UNHEALTHY|
```

Comment:
The same can be here, reasons of marking unstable. Might be useful in the nodeUI or MMN to help users resolve the problem.

Reply:
Also @Snawoot had a point - which states are we still showing in proposals list? I suggest we don't show PENDING and UNHEALTHY